qiskit/tools/qi/pauli.py

```diff
@@ -41,262 +42,487 @@ class Pauli:
-    w = other.w
-    return Pauli(v, w)
-
-
-def label_to_pauli(label):
-    """Return the pauli of a string ."""
-    v = np.zeros(len(label))
-    w = np.zeros(len(label))
-    for j, _ in enumerate(label):
-        if label[j] == 'I':
-            v[j] = 0
-            w[j] = 0
-        elif label[j] == 'Z':
-            v[j] = 1
-            w[j] = 0
-        elif label[j] == 'Y':
-            v[j] = 1
-            w[j] = 1
-        elif label[j] == 'X':
-            v[j] = 0
-            w[j] = 1
+        Raises:
+            QISKitError: when updating whole w, the number of qubits must be the same.
+        """
+        if indices is None:
+            if len(self._w) != len(w):
+                raise QISKitError("During updating whole w, you can not chagne the number of qubits.")
+            self._w = w
         else:
-            print('something went wrong')
-            return -1
-    return Pauli(v, w)
+            if not isinstance(indices, list) or not isinstance(indices, np.ndarray):
+                indices = [indices]
+            for p, idx in enumerate(indices):
+                self._w[idx] = w[p]
+
+        return self
+
+    def insert_qubits(self, indices, pauli_labels):
+        """
+        Insert pauli to the targeted indices.
+
+        Args:
+            indices ([int]): the qubit indices to be inserted.
+            paulis_label([str]): to-be-inserted pauli
 
+        Note:
+            the indices refers to the localion of original paulis,
+            e.g. if indices = [0, 2], pauli_labels = ['Z', 'I'] and original pauli = 'ZYXI'
+            the pauli will be updated to ZY'I'XI'Z'
+            'Z' and 'I' are inserted before the qubit at 0 and 2.
+        """
+        if not isinstance(indices, list):
+            indices = [indices]
+
+        if not isinstance(pauli_labels, list):
+            pauli_labels = [pauli_labels]
+
+        tmp = Pauli.from_label(pauli_labels)
+
+        self._v = np.insert(self._v, indices, tmp.v)
+        self._w = np.insert(self._w, indices, tmp.w)
+
+        return self
+
+    def append_qubits(self, pauli_labels):
+        r"""Append pauli to the higher order of qubit.
+
+        The resulted pauli is $P_{new} \otimes P_{old}$
+
+        Args:
+            paulis_labels(str): to-be-inserted pauli
+
+        Returns:
+            Pauli: self
+        """
+        tmp = Pauli.from_label(pauli_labels)
+        self.kron(tmp)
+
+        return self
 
-def pauli_group(number_of_qubits, case=0):
+    def delete_qubits(self, indices):
+        """
+        Delete pauli at the indices.
+
+        Args:
+            indices([int]): the indices of to-be-deleted paulis.
+
+        Returns:
+            Pauli: self
+        """
+        if not isinstance(indices, list):
+            indices = [indices]
+
+        self._v = np.delete(self._v, indices)
+        self._w = np.delete(self._w, indices)
+
+        return self
+
+    @staticmethod
+    def generate_random_pauli(num_qubits):
```

Comment:
since it does not use any property in the Pauli class, I made it as static method, we can consider it as a classmethod since it is another way to generate a pauli.

Reply:
I don't think methods like `pauli_singles` should be in the class. It would make more sense as a separate function since it is a generator function that returns a list of Pauli objects.